packages/mobile/src/geth/saga.ts

```diff
@@ -177,18 +179,19 @@ function* monitorGeth() {
 }
 
 export function* gethSaga() {
-  yield call(initGethSaga)
-  const gethRelatedSagas = yield fork(monitorGeth)
-  yield take(Actions.CANCEL_GETH_SAGA)
-  yield cancel(gethRelatedSagas)
-  yield put(setGethConnected(true))
-}
-
-export function* gethSagaIfNecessary() {
   yield call(waitForRehydrate) // Wait for rehydrate to know if geth or forno mode
-  yield put(setContractKitReady(true)) // ContractKit is blocked (not ready) before rehydrate
-  if (!(yield select(fornoSelector))) {
-    Logger.debug(`${TAG}@gethSagaIfNecessary`, `Starting geth saga...`)
-    yield call(gethSaga)
+  yield put(setContractKitReady(true)) // TODO(yorke): consider moving elsewhere
+  const fornoMode = yield select(fornoSelector)
+  yield call(initGethSaga)
+  if (!fornoMode) {
+    const gethRelatedSagas = yield fork(monitorGeth)
+    yield take(Actions.CANCEL_GETH_SAGA)
+    yield cancel(gethRelatedSagas)
+  } else {
+    // TODO(yorke): monitor to make sure RPC is still available
+    yield put(setGethConnected(true))
+    yield take(Actions.CANCEL_GETH_SAGA)
   }
+
+  yield put(setGethConnected(false))
```

Comment:
Seems like when we reach here, we should restart or exit the app right? Maybe there's logic elsewhere I'm missing but do we ever recover from this spot and reconnect somehow? Feel free to just leave a TODO